controllers/metal3.io/host_state_machine.go

```diff
@@ -536,30 +536,30 @@ func (hsm *hostStateMachine) handleDeprovisioning(info *reconcileInfo) actionRes
 			hsm.Host.Status.ErrorCount = 0
 		}
 	} else {
-		skipToDelete := func() actionResult {
-			hsm.NextState = metal3api.StateDeleting
+		skipToPowerOff := func() actionResult {
+			hsm.NextState = metal3api.StatePoweringOffBeforeDelete
 			info.postSaveCallbacks = append(info.postSaveCallbacks, deleteWithoutDeprov.Inc)
 			return actionComplete{}
 		}
 
 		switch r := actResult.(type) {
 		case actionComplete:
-			hsm.NextState = metal3api.StateDeleting
+			hsm.NextState = metal3api.StatePoweringOffBeforeDelete
 			hsm.Host.Status.ErrorCount = 0
 		case actionFailed:
 			// If the provisioner gives up deprovisioning and
 			// deletion has been requested, continue to delete.
 			if hsm.Host.Status.ErrorCount > 3 {
 				info.log.Info("Giving up on host clean up after 3 attempts. The host may still be operational " +
 					"and cause issues in your clusters. You should clean it up manually now.")
-				return skipToDelete()
+				return skipToPowerOff()
 			}
 		case actionError:
 			if r.NeedsRegistration() && !hsm.haveCreds {
 				// If the host is not registered as a node in Ironic and we
 				// lack the credentials to deprovision it, just continue to
 				// delete.
-				return skipToDelete()
+				return skipToPowerOff()
```

Comment:
This is kinda useless, you cannot power off without credentials.

Reply:
Done